synapseclient/client.py

```diff
@@ -1337,12 +1337,12 @@ def setAnnotations(self, entity, annotations={}, **kwargs):
 
         :returns: A dictionary
         """
-
-        uri = '/entity/%s/annotations' % id_of(entity)
+        entity = self.get(entity,downloadFile=False)
```

Comment:
I would call restGET(/entity/%s/annotations) instead.  It would be less costly as we don't need the entire entity.